src/libsync/connectionvalidator.cpp

```diff
@@ -119,6 +119,8 @@ void ConnectionValidator::slotStatusFound(const QUrl&url, const QVariantMap &inf
              << url << " with version "
              << CheckServerJob::versionString(info)
              << "(" << CheckServerJob::version(info) << ")";
+    // Newer servers don't disclose any version in status.php anymore
+    // https://github.com/owncloud/core/pull/27473/files
 
     QString version = CheckServerJob::version(info);
     _account->setServerVersion(version);
```

Comment:
This should be guarded by a `if (!version.isEmpty())` - I think we run this periodically and it means that we might reset the server version stored in the account to the empty string occasionally!